### PR TITLE
Added code for modifying connection zones in nmcli module

### LINKED ISSUE #21591: RFE: modify connection zones in nmcli module
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
nmcli module

##### ANSIBLE VERSION
```
devel branch, as of Feb 17 2017
```

##### SUMMARY
The Ansible `nmcli` module has no way to modify the "zone" of an interface.

It would be great if I could do something like this to set the zone on the `docker0` interface:

```yaml
- nmcli:
    conn_name: docker0
    zone: MyZone
```

but the "zone" parameter is not yet supported. I have to run a command like `nmcli connection modify docker0 connection.zone MyZone`

### PR DESCRIPTION
**Pull request by amolkahat**
##### SUMMARY

Added code for modifying connection zones in nmcli module
pep8 fixes in test_nmcli.py file.
Fixes: #21591

Signed-off-by: Amol Kahat <akahat@redhat.com>

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
nmcli

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel)
```

### CODE REVIEW

File: test/units/modules/net_tools/test_nmcli.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Copyright: (c) 2017 Ansible Project
-# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```
Comment (Akasurde) [inline test/units/modules/net_tools/test_nmcli.py]: No need of the new line here.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: test/units/modules/net_tools/test_nmcli.py [hunk 2]
```diff
@@ -152,7 +187,8 @@ def mocker_set(mocker, connection_exists=False):
     """
     mocker.patch('ansible.modules.net_tools.nmcli.HAVE_DBUS', True)
     mocker.patch('ansible.modules.net_tools.nmcli.HAVE_NM_CLIENT', True)
-    get_bin_path = mocker.patch('ansible.module_utils.basic.AnsibleModule.get_bin_path')
+    get_bin_path = mocker.patch(
+        'ansible.module_utils.basic.AnsibleModule.get_bin_path')
```
Comment (oussemos) [inline test/units/modules/net_tools/test_nmcli.py:191]: The actual `max-line-length` is 160, so there's no need of a new lines here 

https://github.com/ansible/ansible/blob/379a7f4f5a0491964c7896834f3f326412888585/tox.ini#L31

**Review by oussemos — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@oussemos Would you be interested in taking over this PR ? Let me know.

**Comment by oussemos** [comment]
ok @Akasurde

**Comment by Akasurde** [comment]
superseded by #48672

**Pull request closed without merging**